pyomo/core/plugins/transform/scaling.py

```diff
@@ -313,10 +313,13 @@ def propagate_solution(self, scaled_model, original_model):
             original_v = original_model.find_component(original_v_path)
 
             for k in scaled_v:
-                original_v[k].set_value(
-                    value(scaled_v[k]) / component_scaling_factor_map[scaled_v[k]],
-                    skip_validation=True,
-                )
+                if scaled_v[k].value is not None:
```

Comment:
Is the correct check here to test `None`, or to not transfer the values of any `stale` variables?  I am wondering if the correct answer is to only propagate variable values when `.stale==False`

Reply:
I guess I can see three options:
1. Propagate values in the model, unless they are None (what this PR currently does)
2. Propagate values in the model, regardless of what they are
3. Propagate non-stale values in the model

My instinct is that relying on the stale flag can lead to non-intuitive behavior, e.g. if a user sets a value after a solve (or solves via `solve_strongly_connected_components`), only the last values updated will be propagated. If this is what we decide on, the docstring should be updated to indicate that only the most recent solution is propagated.

Reply:
Based on @dallan-keylogic's comment on #2817, I've updated this PR to propagate values from the scaled model even if they are None, but to log a warning if None overrides a non-None value.

Reply:
I think I am OK with this behavior, but if we are always going to propagate the entire model state, I wonder if a better name for this method is `propagate_state()` (as we are propagating the current state of the model and not necessarily the last thing that came back from the solver).

Reply:
I agree that `propagate_state`, or something like `propagate_variable_values`, would be a better name, although I'm not convinced it's worth changing.